openlibrary/catalog/marc/parse.py

```diff
@@ -668,6 +687,12 @@ def read_edition(rec):
         update_edition(rec, edition, read_languages, 'languages')
         update_edition(rec, edition, read_pub_date, 'publish_date')
 
+    saved_language = edition['languages'][0] if 'languages' in edition else None
```

Comment:
I would like this line to not be necessary, but I imagine it breaks the current tests to remove it?

A unified `008` / `041` language extraction `read_languages()` would handle this better, but I'm imagining that this a re-factoring task for later.

Reply:
It's needed because of the odd way that `update_edition()` is structured, where the function can't have any parameters, but I think struggling to get the `041` parsing working may have caused me to abandon a more direct solution too early. 

Let me see if I can simplify things.

Reply:
This whole section is gone and I've refactored read_languages to take the `008` language as input and generate a single set of languages.